src/core/positioning.cpp

```diff
@@ -0,0 +1,203 @@
+/***************************************************************************
+  positioning.cpp - Positioning
+
+ ---------------------
+ begin                : 22.05.2022
+ copyright            : (C) 2022 by Mathieu Pellerin
+ email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "bluetoothreceiver.h"
+#include "internalreceiver.h"
+#include "positioning.h"
+#include "positioningutils.h"
+
+Positioning::Positioning( QObject *parent )
+  : QObject( parent )
+{
+}
+
+void Positioning::setActive( bool active )
+{
+  if ( mActive == active )
+    return;
+
+  mActive = active;
+
+  if ( mActive )
+  {
+    if ( !mReceiver )
+    {
+      setupDevice();
+    }
+    mReceiver->connectDevice();
+  }
+  else
+  {
+    if ( mReceiver )
+    {
+      mReceiver->disconnectDevice();
+    }
+  }
+
+  emit activeChanged();
+}
+
+void Positioning::setDevice( const QString &device )
+{
+  if ( mDevice == device )
+    return;
+
+  mDevice = device;
+  setupDevice();
+
+  emit deviceChanged();
+}
+
+void Positioning::setValid( bool valid )
+{
+  if ( mValid == valid )
+    return;
+
+  mValid = valid;
+
+  emit validChanged();
+}
+
+void Positioning::setAveragedPosition( bool averaged )
+{
+  if ( mAveragedPosition == averaged )
+    return;
+
+  mAveragedPosition = averaged;
+  if ( mAveragedPosition )
+  {
+    mCollectedPositionInformations << mPositionInformation;
+  }
+  else
+  {
+    mCollectedPositionInformations.clear();
+  }
+
+  emit averagedPositionCountChanged();
+  emit averagedPositionChanged();
+}
+
+void Positioning::setCoordinateTransformer( QgsQuickCoordinateTransformer *coordinateTransformer )
+{
+  if ( mCoordinateTransformer == coordinateTransformer )
+    return;
+
+  if ( mCoordinateTransformer )
```

Comment:
Do we need to empty averaged points in here (or are they stored unprojected)?

Reply:
@m-kuhn , stored unprojected.